server/src/lib/model/clips.ts

```diff
@@ -145,10 +152,36 @@ export default class Clips {
     clip.sentenceText = text;
   }
 
+  private clipsWithFewVotes: DBClipWithVoters[] = [];
+  private isRefilling = false;
+  private async refillClipsWithFewVotes() {
+    if (this.isRefilling) return;
+    this.isRefilling = true;
+    this.clipsWithFewVotes = this.clipsWithFewVotes.concat(
+      Random.shuffle(
+        this.randomEngine,
+        await this.db.findClipsWithFewVotes(CLIP_CACHE_SIZE)
+      )
+    );
+    this.isRefilling = false;
+  }
+
   /**
    * Fetch a random clip but make sure it's not the user's.
    */
-  getEllibleClip(userid: string): Clip {
+  async getEllibleClip(client_id: string): Promise<DBClipWithVoters | Clip> {
+    if (this.isMigrated) {
+      if (!this.clipsWithFewVotes.length) await this.refillClipsWithFewVotes();
+
+      const i = this.clipsWithFewVotes.findIndex(
+        clip => clip.client_id !== client_id && !clip.voters.includes(client_id)
+      );
+      if (i == -1) return null;
```

Comment:
nit: please use strong equality whenever possible

Reply:
I'd say when using TS it's fine for comparing literals (because then you can be sure that you're comparing literals). We could add an ESLint rule for that.